### PR TITLE
MT Task in SpeechLM

### PR DESCRIPTION
**Pull request by ftshijt**
The full MT-task implementation for speechlm.

### CODE REVIEW

File: espnet2/bin/speechlm_inference.py [hunk 1]
```diff
@@ -381,7 +381,8 @@ def inference(
                         logging.info(f"Save audio: {audio_path}")
                     
                     elif modality in ["text_bpe"]:
-                        writers[name].write(f"{example_name} {detokenized}")
+                        detokenized = detokenized.strip()
```
Comment (ftshijt) [inline espnet2/bin/speechlm_inference.py:384]: @jctian98 Please let me know if this is fixed in your latest PR then I can remove this temporary modification.

File: egs2/TEMPLATE/speechlm1/speechlm.sh [hunk 2]
```diff
@@ -822,7 +825,7 @@ if ! "${skip_eval}"; then
                 --nj ${nj} \
                 --inference_nj ${inference_nj} \
                 --gpu_inference ${gpu_inference} \
-                --nbest ${nbest}
+                --nbest ${nbest} ${scoring_args}
```
Comment (ftshijt) [inline egs2/TEMPLATE/speechlm1/speechlm.sh:826]: @jctian98 I intentionally add this one to incorporate scoring related arguments for different tasks.

**Comment by ftshijt** [comment]
@jctian98 This is a following up fix for MT task of speechlm project, mostly with evaluation.

**Comment by jctian98** [comment]
it's ok to merge this to speechlm branch for now. I'll take care of the compatibility issues.

**Pull request merged**